extensions/amp-image-slider/1.0/component.js

```diff
@@ -0,0 +1,701 @@
+    labels,
+    shouldHintReappear,
+    stepSize = 0.1,
+    ...rest
+  },
+  ref
+) {
+  /** Common variables */
+  let containsAmpImages;
+  const gesturesRef = useRef(null);
+  const {isEdge, setIsEdge} = useState();
+  const isEventRegistered = false;
+  const styles = useStyles();
+
+  /** Common variables */
+  const win = useRef(null);
+  const doc = useRef(null);
+
+  /** Container Reference */
+  const containerRef = useRef(null);
+
+  /** Image References */
+  const leftImageRef = useRef(null);
+  const rightImageRef = useRef(null);
+
+  /** Label References */
+  const leftLabelWrapperRef = useRef(null);
+  const leftLabelRef = useRef(null);
+  const rightLabelWrapperRef = useRef(null);
+  const rightLabelRef = useRef(null);
+
+  /** Mask References */
+  const leftMaskRef = useRef(null);
+  const rightMaskRef = useRef(null);
+
+  /** Bar Reference */
+  const barRef = useRef(null);
+
+  /** Arrow References */
+  const leftHintArrowRef = useRef(null);
+  const rightHintArrowRef = useRef(null);
+
+  /** Hint Body References */
+  const leftHintBodyRef = useRef(null);
+  const rightHintBodyRef = useRef(null);
+
+  /** Unlisten Handlers for Mouse */
+  const unlistenMouseUp = useRef(null);
+  const unlistenMouseDown = useRef(null);
+  const unlistenMouseMove = useRef(null);
+
+  /** Unlisten Handlers for Keyboard */
+  const unlistenKeyDown = useRef(null);
+
+  const buildImageWrappers = useCallback(() => {
+    leftMaskRef.current = createElementWithAttributes(doc.current, 'div');
+    rightMaskRef.current = createElementWithAttributes(doc.current, 'div');
+    containerRef.current.appendChild(leftMaskRef.current);
+    containerRef.current.appendChild(rightMaskRef.current);
+
+    leftMaskRef.current.classList.add('i-amphtml-image-slider-left-mask');
+
+    if (leftLabelRef.current != undefined) {
+      leftLabelWrapperRef.current = createElementWithAttributes(
+        doc.current,
+        'div'
+      );
+      leftLabelWrapperRef.current.classList.add(
+        'i-amphtml-image-slider-label-wrapper'
+      );
+      leftLabelWrapperRef.current.appendChild(leftLabelRef.current);
+      leftMaskRef.current.appendChild(leftLabelWrapperRef.current);
+    }
+
+    rightMaskRef.current.classList.add('i-amphtml-image-slider-right-mask');
+    rightMaskRef.current.classList.add('i-amphtml-image-slider-push-right');
+    rightImageRef.current.classList.add('i-amphtml-image-slider-push-left');
+    if (rightLabelRef.current != undefined) {
+      rightLabelWrapperRef.current = createElementWithAttributes(
+        doc.current,
+        'div'
+      );
+      rightLabelWrapperRef.current.classList.add(
+        'i-amphtml-image-slider-label-wrapper'
+      );
+
+      rightLabelWrapperRef.current.classList.add(
+        'i-amphtml-image-slider-push-left'
+      );
+      rightLabelWrapperRef.current.appendChild(rightLabelRef.current);
+      rightMaskRef.current.appendChild(rightLabelWrapperRef.current);
+    }
+  }, []);
```

Comment:
Preact component should not be creating any elements with DOM APIs. Please prefer JSX instead, i.e.

```js
const leftMaskRef = createElementWithAttributes(doc.current, 'div')
const rightMaskRef = createElementWithAttributes(doc.current, 'div')
```

should instead be:
```jsx
return <div>
  <div ref={leftMaskRef} class={classes.leftMask)></div>
  <div ref={rightMaskRef} class={classes.rightMask)></div>
</div>
```

Please update all uses of `createElement`, `appendChild`, `classList.add` with the above declarative notation.